### ISSUE TITLE
SDK 9.1 requires "web" scope whereas 9.0 did not

### ISSUE THREAD

**Issue by ferrerod** [forcedotcom/SalesforceMobileSDK-iOS#3406]
Please fill out the following details:

1. Version of Mobile SDK Used: 9.1.0 and 9.1.1
2. Issue found in Native App or Hybrid App: Native iOS
3. OS Version: iOS 14.x
4. Device: iPad, iPad Simulator
5. Steps to reproduce: iOS native (Connected app) passing "api" in bootconfig.plist - upon successful credentials provided app fails login.
6. Actual behavior: 
valid credentials results in failure with error: error=OAUTH_DISALLOWED_SCOPES&error_description=the+requested+scope+is+not+allowed&display=touch"

7. Expected Behavior: valid credentials should result in successful login
8. Error Log:

Using SF SDK 9.0.0 pod, we were successfully logging into the Connected App with simply "api" in the bootconfig.plist but with 9.1.0 and 9.1.1 version of the pod, we can only login successfully if we add "web" to the bootconfig.plist

Looking at the changes to SF SDK source between 9.0.0 and 9.1.0 I see hybridToken is now used.

in 9.1.1 SFOAuthCoordinator.m: 658
changed from:
    [approvalUrlString appendFormat:@"&%@=%@", kSFOAuthResponseType, kSFOAuthResponseTypeToken];
changed to:
(    [approvalUrlString appendFormat:@"&%@=%@", kSFOAuthResponseType, kSFOAuthResponseTypeHybridToken];

I don't know if the above change is the problem or that the RemoteAccessAuthorizationPage.apexp is the problem. The login flow errors out when this page is requested:
.../setup/secur/RemoteAccessAuthorizationPage.apexp?display=touch&source=xxxxxxxxxx

and the error is 
error=OAUTH_DISALLOWED_SCOPES&error_description=the+requested+scope+is+not+allowed&display=touch"

**Comment by ferrerod** [comment]
I should note, the connected app, in the connected app oath settings does list several permissions:

- Perform requests on your behalf at any time
- Access your basic information
- Provide access to your data via the web
- Access and manage your data

perhaps the problem is that the portal config seems to require web but the bootconfig.plist does not pass web?

**Comment by bhariharan** [comment]
Typically, the scopes in `bootconfig` should match the scopes in the connected app interface. Even though `web` wasn't a hard requirement in the past, it was required for some flows such as "Advanced Authentication" to function properly. This is why we have included `web` as a scope in all our templates, including the native ones. In `9.1`, we have upgraded our OAuth flow to a better version that allows more granular control and introduces a few extra scopes. This new flow requires `web` to function properly. This is by design.
